ompi/mca/coll/base/coll_base_bcast.c

```diff
@@ -236,19 +235,19 @@ ompi_coll_base_bcast_intra_generic( void* buffer,
         if (err != MPI_SUCCESS) { line = __LINE__; goto error_hndl; }
     }
 
-#if !defined(COLL_BASE_BCAST_USE_BLOCKING)
-    if( NULL != send_reqs ) free(send_reqs);
-#endif
-
     return (MPI_SUCCESS);
 
  error_hndl:
     OPAL_OUTPUT( (ompi_coll_base_framework.framework_output,"%s:%4d\tError occurred %d, rank %2d",
                   __FILE__, line, err, rank) );
-#if !defined(COLL_BASE_BCAST_USE_BLOCKING)
-    if( NULL != send_reqs ) free(send_reqs);
-#endif
-    return (err);
+    if( MPI_SUCCESS != err ) {
+        ompi_coll_base_free_reqs( recv_reqs, 2);
+        if( NULL != send_reqs ) {
+            ompi_coll_base_free_reqs( send_reqs, tree->tree_nextsize);
```

Comment:
@bosilca  Restore the #if around send_reqs references: send_reqs is defined
only if #if !defined(COLL_BASE_BCAST_USE_BLOCKING)
Actually, I didn't find the place where this COLL_BASE_BCAST_USE_BLOCKING is defined
Maybe it should definitely be removed?